sklearn/decomposition/_nmf.py

```diff
@@ -903,6 +903,10 @@ def non_negative_factorization(X, W=None, H=None, n_components=None, *,
 
         - None: 'nndsvd' if n_components < n_features, otherwise 'random'.
 
+        .. deprecated:: 0.24
+           ``nndsvd`` has been deprecated for ``solver='mu'`` in version 0.24
```

Comment:
This is not exactly what we are doing.
We are not deprecating "nndsvd", we are changing the behavior of `init=None`.